dspace-server-webapp/src/main/java/org/dspace/app/rest/signposting/service/impl/LinksetServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+ * Default implementation of {@link LinksetService}.
+ */
+@Service
+public class LinksetServiceImpl implements LinksetService {
+
+    private static final Logger log = Logger.getLogger(LinksetServiceImpl.class);
+
+    @Autowired
+    private BitstreamService bitstreamService;
+
+    @Autowired
+    private BitstreamMetadataReadPermissionEvaluatorPlugin bitstreamMetadataReadPermissionEvaluatorPlugin;
+
+    private final List<BitstreamSignpostingProcessor> bitstreamProcessors = new DSpace().getServiceManager()
+            .getServicesByType(BitstreamSignpostingProcessor.class);
+
+    private final List<ItemSignpostingProcessor> itemProcessors = new DSpace().getServiceManager()
+            .getServicesByType(ItemSignpostingProcessor.class);
+
+    private final List<MetadataSignpostingProcessor> metadataProcessors = new DSpace().getServiceManager()
+            .getServicesByType(MetadataSignpostingProcessor.class);
+
+    @Override
+    public List<List<LinksetNode>> createLinksetNodesForMultipleLinksets(
+            HttpServletRequest request,
+            Context context,
+            Item item
+    ) {
+        ArrayList<List<LinksetNode>> linksets = new ArrayList<>();
+        addItemLinksets(request, context, item, linksets);
+        addBitstreamLinksets(request, context, item, linksets);
+        addMetadataLinksets(request, context, item, linksets);
+        return linksets;
+    }
+
+    @Override
+    public List<LinksetNode> createLinksetNodesForSingleLinkset(
+            HttpServletRequest request,
+            Context context,
+            DSpaceObject object
+    ) {
+        List<LinksetNode> linksetNodes = new ArrayList<>();
+        if (object.getType() == Constants.ITEM) {
+            for (ItemSignpostingProcessor processor : itemProcessors) {
+                processor.addLinkSetNodes(context, request, (Item) object, linksetNodes);
+            }
+        } else if (object.getType() == Constants.BITSTREAM) {
+            for (BitstreamSignpostingProcessor processor : bitstreamProcessors) {
+                processor.addLinkSetNodes(context, request, (Bitstream) object, linksetNodes);
+            }
+        }
+        return linksetNodes;
+    }
+
+    private void addItemLinksets(
+            HttpServletRequest request,
+            Context context,
+            Item item,
+            List<List<LinksetNode>> linksets
+    ) {
+        List<LinksetNode> linksetNodes = new ArrayList<>();
+        if (item.getType() == Constants.ITEM) {
+            for (ItemSignpostingProcessor sp : itemProcessors) {
+                sp.addLinkSetNodes(context, request, item, linksetNodes);
+            }
+        }
+        linksets.add(linksetNodes);
+    }
+
+    private void addBitstreamLinksets(
+            HttpServletRequest request,
+            Context context,
+            Item item,
+            ArrayList<List<LinksetNode>> linksets
+    ) {
+        Iterator<Bitstream> bitstreamsIterator = getItemBitstreams(context, item);
```

Comment:
@NikitaKr1vonosov and @abollini : This appears to be adding a linkset for all Bitstreams of an Item, regardless of the Bundle.  Is that expected behavior?  Currently, I'm seeing Bitstreams in TEXT and THUMBNAIL and LICENSE bundles listed when I use `[dspace.ui.url]/signposting/linksets/[item-uuid]/json`

Reply:
fixed